### PR TITLE
compiler: Block reductions irrespective of par-tile

### PR DESCRIPTION
**Pull request by FabioLuporini**
Test is in PRO

I also took the chance to create opt-options to allow specifying different par-tiles for reduction loops

### CODE REVIEW

File: devito/passes/clusters/blocking.py [hunk 1]
```diff
@@ -447,14 +449,42 @@ def __init__(self, par_tile):
...
         x = any(i.dim.is_Block for i in flatten(c.ispace for c in clusters))
         if not x:
             self.tip += 1
 
-        # TODO: This is for now exceptionally rudimentary
+        # Correctness -- enforce blocking where necessary.
+        # See also issue #276:PRO
+        if any(c.properties.is_parallel_atomic(d) for c in clusters):
+            if any(c.is_sparse for c in clusters):
+                if not x:
```
Comment (mloubout) [inline devito/passes/clusters/blocking.py:479]: Don't think you don't need this, it's just a tuple not a multi-tuple, same below

**Review by mloubout — COMMENTED** [review]
Small comment, gtg

File: devito/passes/clusters/blocking.py (hunk 1 above)
Comment (mloubout) [reply devito/passes/clusters/blocking.py:479]: Wait no you do need it to go back to first element if there is multiple sparse/reduce cluster nevermind

Comment (FabioLuporini) [reply devito/passes/clusters/blocking.py:479]: correct

**Pull request merged**